docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md

```diff
@@ -0,0 +1,327 @@
+         ]
+      }
+   ]
+})
+}
+```
+
+To assign the policy to a role for the IAM role to service account mapping in EKS, a Terraform module like [this one](https://registry.terraform.io/modules/terraform-aws-modules/iam/aws/latest/submodules/iam-role-for-service-accounts-eks) is helpful.
+
+```json
+module "aurora_role" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  role_name = "aurora-role"
+
+  role_policy_arns = {
+    policy = aws_iam_policy.rds_policy.arn
+  }
+
+  oidc_providers = {
+    main = {
+      provider_arn               = "arn:aws:iam::account-id:oidc-provider/oidc.eks.region.amazonaws.com/id/eks-id"
+      namespace_service_accounts = ["aurora-namespace:aurora-serviceaccount"]
+    }
+  }
+}
+```
+
+These two Terraform snippets allow the service account `aurora-serviceaccount` within the `aurora-namespace` to assume the user `db-user-name` within the database `DbiResourceId`.
+The output of the module `aurora_role` has the output `iam_role_arn` to annotate a service account to make use of the mapping.
+
+Annotate the service account with the `iam_role_arn` output of the `aurora_role`.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::account-id:role/role-name
+  name: aurora-serviceaccount
+  namespace: aurora-namespace
+```
+
+The setup required on the Aurora Postgres side is to actually create the user and assign the required permissions to it. The following is an example when connected to the Postgres database, and can also be realized differently. See the [AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html#UsingWithRDS.IAMDBAuth.DBAccounts.PostgreSQL) for reference.
+
+```SQL
+# create user and grant rds_iam role, which requires the user to login via IAM authentication over password
+CREATE USER "db-user-name";
+GRANT rds_iam TO "db-user-name";
+
+# create some database and grant the user all privileges to it
+CREATE DATABASE "some-db";
+GRANT ALL privileges on database "some-db" to "db-user-name";
+```
+
+### Keycloak
+
+For Keycloak versions 21+, the default JDBC driver can be overwritten, allowing use of a custom wrapper like the [aws-advanced-jdbc-wrapper](https://github.com/awslabs/aws-advanced-jdbc-wrapper) to utilize the features of IRSA. This is a wrapper around the default JDBC driver, but takes care of signing the requests.
+
+The following example uses the mentioned `aws-advanced-jdbc-wrapper`. Additionally, see the [Keycloak documentation](https://www.keycloak.org/server/db#_overriding_the_default_jdbc_driver) on overwriting the default JDBC driver.
+
+A custom Docker image is required as there's currently no upstream image with all the configurations required.
+
+Required are the following artifacts for the `aws-advanced-jdbc-wrapper` to work:
+
+- https://mvnrepository.com/artifact/software.amazon.awssdk/regions
```

Comment:
I'm trying to think of a way to make this more clean. Could we instead provide [words]() with the links supplied to trim it up?

Reply:
good idea! I'll switch it up to just be the artifact name and also mention prior that all of those artifacts are from `software.amazon.awssdk` just to make sure there's no misunderstanding.